src/storage/src/table/state_table.rs

```diff
@@ -239,76 +260,71 @@ where
         let cell_based_table_iter = self.cell_based_table_iter.peekable();
```

Comment:
Also fuse this stream?
```suggestion
        let cell_based_table_iter = self.cell_based_table_iter.fuse().peekable();
```

Reply:
+1